tests/test_hf_api.py

```diff
@@ -780,6 +788,14 @@ def test_filter_models_with_complex_query(self):
             ["pytorch" in model.tags and "tf" in model.tags for model in models]
         )
 
+    @with_production_testing
+    def test_filter_models_with_cardData(self):
+        _api = HfApi()
+        models = _api.list_models("co2_eq_emissions", cardData=True)
+        self.assertTrue([hasattr(model, "cardData") for model in models])
+        models = _api.list_models("co2_eq_emissions", cardData=False)
+        self.assertTrue([not hasattr(model, "cardData") for model in models])
```

Comment:
What is this test's aim? From my local tests, I see that `list_models("co2_eq_emissions", cardData=False)` does return models with `cardData`.

The assertion checks that the list (which is made up of all `False` values) is truthy, which it is because a list is truthy as long as it has length.

I would expect a test such as the following:
```suggestion
        models = _api.list_models("co2_eq_emissions", cardData=False)
        self.assertTrue(all([not hasattr(model, "cardData") for model in models]))
```

But that would fail right now

Reply:
That actually seems to be an issue with the backend API, no? Because the assumption should be that calling with `cardData:False` should not return the `cardData`, as @julien-c noted here: https://github.com/huggingface/moon-landing/issues/1584#issuecomment-991344206

But this is actually inconsistent, telling me that something might be wrong with our backend. Why? When I do:
```python
o = api.list_models(cardData=False, full=False)
i = 0
models = []
for model in o:
    if hasattr(model, 'cardData'): 
        i+= 1
        models.append(model)
keys = set()
non_dict = []
for model in models: 
    if isinstance(model.cardData, dict):
        for key in list(model.cardData.keys()):
            keys.add(key)
    else:
        non_dict += [model]
```
What I discover is that there are models that return a modelInfo with a `co2_emissions` key, even though that should only happen when we set `cardData` to `True`. Along with this, there is a broken model that has the wrong card data, it's a list rather than a dictionary. Is that expected?

(This is also shown when I do this via a raw `requests`)

Reply:
@julien-c this may also tie into your thinking on how they should have very different content sizes

Reply:
not sure, on my side i was just mentioning that `wget https://huggingface.co/api/models` and `wget https://huggingface.co/api/models?cardData=true` have 2x size diff